File: backend/Mover.Entities/POCOEntities/Equipo.cs

using Mover.Entities.POCOEntities.Vehiculos;

namespace Mover.Entities.POCOEntities
{
    public class Equipo
    {
        public int Id { get; set; }        
        public int UserId { get; set; }        
        public string PlacasVehiculo { get; set; } = string.Empty;
        public int MarcaId { get; set; }
        public int ModeloId { get; set; }        
        public int ReferenciaId { get; set; }        
        public int CarroceriaId { get; set; }        
        public string Color { get; set; } = string.Empty;
        public string PlacasSemirremolque { get; set; } = string.Empty;
        public string LinkLicenciaConduccion { get; set; } = string.Empty;
        public string LinkSoat { get; set; } = string.Empty;
        public string LinkTecnomecanica { get; set; } = string.Empty;
        public int? EstadoEquipoId { get; set; }


        public User? User { get; set; }
        public EstadoEquipo? EstadoEquipo { get; set; }
        public Marca? Marca { get; set; }
        public Modelo? Modelo { get; set; }
        public Referencia? Referencia { get; set; }
        public Carroceria? Carroceria { get; set; }

    }
}

File: frontend/lib/domain/entities/equipo.dart
class Equipo {
  final int? id;
  final int? userId;
  final String placasVehiculo;
  final int? marcaId;
  final int? modeloId;
  final int? referenciaId;
  final int? carroceriaId;
  final String color;
  final String placasSemirremolque;
  final String linkLicenciaConduccion;
  final String linkSoat;
  final String linkTecnomecanica;
  final int? estadoEquipoId;

  Equipo({
    this.id,
    this.userId,
    this.placasVehiculo = "",
    this.marcaId,
    this.modeloId,
    this.referenciaId,
    this.carroceriaId,
    this.color = "",
    this.placasSemirremolque = "",
    this.linkLicenciaConduccion = "",
    this.linkSoat = "",
    this.linkTecnomecanica = "",
    this.estadoEquipoId,
  });

  /// Método para crear un objeto `Equipo` desde un JSON.
  factory Equipo.fromJson(Map<String, dynamic> json) {
    return Equipo(
      id: json['id'],
      userId: json['userId'],
      placasVehiculo: json['placasVehiculo'] ?? "",
      marcaId: json['marcaId'],
      modeloId: json['modeloId'],
      referenciaId: json['referenciaId'],
      carroceriaId: json['carroceriaId'],
      color: json['color'] ?? "",
      placasSemirremolque: json['placasSemirremolque'] ?? "",
      linkLicenciaConduccion: json['linkLicenciaConduccion'] ?? "",
      linkSoat: json['linkSoat'] ?? "",
      linkTecnomecanica: json['linkTecnomecanica'] ?? "",
      estadoEquipoId: json['estadoEquipoId'],
    );
  }

  /// Método para convertir un objeto `Equipo` a JSON.
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'placasVehiculo': placasVehiculo,
      'marcaId': marcaId,
      'modeloId': modeloId,
      'referenciaId': referenciaId,
      'carroceriaId': carroceriaId,
      'color': color,
      'placasSemirremolque': placasSemirremolque,
      'linkLicenciaConduccion': linkLicenciaConduccion,
      'linkSoat': linkSoat,
      'linkTecnomecanica': linkTecnomecanica,
      'estadoEquipoId': estadoEquipoId,
    };
  }
}
